src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -2861,16 +2866,16 @@ def import_zone(resource_group_name, zone_name, file_name):
                     if record_set_type != 'soa' and relative_record_set_name != origin:
                         relative_record_set_name = record_set_name[:-(len(origin) + 2)]
 
-                    record_set = RecordSet(
-                        name=relative_record_set_name, type=record_set_type, ttl=record_set_ttl)
+                    record_set = RecordSet(ttl=record_set_ttl)
                     record_sets[record_set_key] = record_set
                 _add_record(record_set, record, record_set_type,
                             is_list=record_set_type.lower() not in ['soa', 'cname'])
 
     total_records = 0
-    for rs in record_sets.values():
+    for key, rs in record_sets.items():
+        rs_type = key.rsplit('.', 1)[1].lower()
```

Comment:
Ignore the comment if it is over-concern, but `[-1]` would be safer if `.` could be optional

Reply:
Well, I'd still have to split on the '.' to use that since key is just a string.